### ISSUE TITLE
packages folder is created while storage: none in paket.dependencies file

### ISSUE THREAD

**Issue by moh-hassan** [fsprojects/Paket#3423]
This issue is opened as a request for an [issue ](https://github.com/fsprojects/Paket/issues/3388#issuecomment-437765003)

Paket version: 5.186.3
I applied the instruction described [here ](https://fsprojects.github.io/Paket/dependencies-file.html#Disable-packages-folder) for Disabling packages folder in SDK project style.

I tried the project  ([source code](https://github.com/fsprojects/Paket/files/2489612/depresolution-4.zip)) with modifying the paket.dependencies file by adding:

            storage: none 

Paket use nuget cache properly and all packages are in cache but the  package folder is created in solution folder when i run:

           paket restore -v

Part of paket logs with unzipping packages:

>Starting full restore process.
Castle.Core 4.3.1 already downloaded.
Castle.Core 4.3.1 unzipped to F:\projects\project_issues\depresolution-4\depresolution\packages\Castle.Core
Microsoft.AspNet.WebApi.Client 5.2.6 already downloaded.
Microsoft.AspNet.WebApi.Client 5.2.6 unzipped to F:\projects\project_issues\depresolution-4\depresolution\packages\Microsoft.AspNet.WebApi.Client

I think that paket should avoid unzipping and creating `**package**` folder to avoid consuming extra storage.
All packages are already unzipped in the cache, so why unzipping again?
What I missed to disable creating package folder?

**Comment by inosik** [comment]
You have to run `install` or `update` to update your `paket.lock` file. Paket actually emits a warning:

``` shell
$ paket restore
Checking Paket version (version 5.173.1 requested)...
Paket.exe 5.173.1 is up to date.
Paket version 5.173.1
paket.dependencies and paket.lock are out of sync in G:\projects\experiments\paket\3423-packages-folder-with-storage-none.
Please run 'paket install' or 'paket update' to recompute the paket.lock file.
Changes were detected for Main/Castle.Core
    - SettingsChanged
...
```

**Comment by moh-hassan** [comment]
In a previous run, I have already run `paket install` then `paket restore` and all packages are downloaded to nuget cache.

I deleted paket-files folder , delete paket.lock and disabled network card to stop downloading from internet

run:

            paket install

I expect paket install from nuget cache, but I get error:

>.paket\paket install -v
Paket version 5.186.3
found: F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies
Parsing F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies
Resolving packages for group Main:
   0 packages in resolution.
   6 requirements left
     - Microsoft.AspNet.WebApi.Client,  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)
     - System.Data.SqlClient,  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)
     - Microsoft.NET.Test.Sdk,  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)
     - Moq,  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)
     - NUnit,  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)
     - NUnit3TestAdapter,  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)

>  Trying to resolve Microsoft.AspNet.WebApi.Client  (from F:\projects\project_issues\depresolution-4\depresolution\paket.dependencies)
Starting request to 'https://api.nuget.org/v3/index.json'
Performance:
> - Resolver: 293 milliseconds (1 runs)
>    - Runtime: 212 milliseconds
 >   - Blocked (retrieving package versions): 80 milliseconds (1 times)
 >- Average Request Time: 19 milliseconds
 >- Number of Requests: 1
 >- Runtime: 909 milliseconds
>Paket failed with
>-> Unable to retrieve package versions for 'Microsoft.AspNet.WebApi.Client'
>   -- CLOSED --

I enabled network card, run paket install
paket  downloaded again 14 packages

**I expect:**

When run paket install

`paket Install` can work without enabling network card and connect to internet as if all packages and its dependency are in nuget cache

paket check packages in the nuget cache, build dependency tree from cache(not nuget org via internet), update paket.lock and raise error if package/or dependency not available.

**Actual:**

it have to connect to internet to build dependency tree  and is not  computing dependency from cache and raise exception error as described above.

**Comment by forki** [comment]
Paket install is a command that needs internet. It was never designed to be
run without it.

The only command that should run completely fine without internet is
restore - given package cache is already populated.

Am Mi., 14. Nov. 2018, 19:21 hat Mohamed Hassan <notifications@github.com>
geschrieben:

**Comment by moh-hassan** [comment]
Paket restore can't work without paket install and paket install need connection to internet even the cache include all packages and its dependency.

Some Packages in cache may not be available in nuget.org in internet.
The command [  nuget add ](https://docs.microsoft.com/en-us/nuget/tools/cli-ref-add)  can add a package in offline repository then `dotnet restore or msbuil /t:restore ` cache the package in nuge cache.

How i do that behavior using paket.

**Comment by forki** [comment]
No I think you are mixing different things here. Install needs connection to check which package versions are available. We do not get that information from the cache.

**Comment by moh-hassan** [comment]
>I think you are mixing different things here.

No, i didn't mix different things.
The changes in nuget, msbuild and dotnet technology agree all in respecting the nuget cache.

I can restore and build without connecting to internet as if all packages and dependency are exist in cache.

All dependency information are exist in the nuspec file included in the package.

Any way, I understand `storage:none` for a new project , that i can re-use nuget cashe.

**Comment by forki** [comment]
No. The install process is something that does not exist as such in vanilla nuget. It's active resolution process that tries to discover latest versions that fit your dependencies file. It's by design that it uses internet connection. And it's something that that always runs in a user context, while restore build etc can run in CI context.

**Comment by moh-hassan** [comment]
Thanks @forki  for clarification the install process for discovering the  latest versions and why internet needed.
